### PR TITLE
DS-3768 Fixes the harvest solr parse error by 

### PR DESCRIPTION
**Pull request by marsaoua**
setting the requirements and syntaxes of the field "location" and "lastModified" properly.

https://jira.duraspace.org/browse/DS-3768

### CODE REVIEW

**Comment by pnbecker** [comment]
Should also be cherry-picked to master.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java [hunk 1]
```diff
@@ -267,17 +269,23 @@ else if ( ! itemsChanged(context, dso, cFeed.timeStamp))
...
+         * of the form 1995-12-31T23:59:59Z The trailing "Z" designates UTC time
+         * and is mandatory (See below for an explanation of UTC). Optional
+         * fractional seconds are allowed, as long as they do not end in a
+         * trailing 0 (but any precision beyond milliseconds will be ignored).
+         * All other parts are mandatory." Therefore the Full ISO 8601 format
+         * is e.g. "2009-07-16T13:59:21Z" is maintained
+         */
         DCDate dcStartDate = new DCDate( new Date(timeStamp) );
         DCDate dcEndDate = new DCDate( new Date(System.currentTimeMillis()) );
-
```
Comment (pnbecker) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: @tdonohue I have in mind that you was juggling with this sometime ago. Right? Can you please take a look?

Comment (tdonohue) [reply dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: No, I don't recall working on this recently.  That said, the code here looks reasonable enough to me..but likely needs testing, as there seem to be significant changes to the Solr queries (and I'll admit, I'm not familiar with these Solr queries myself).

Comment (pnbecker) [reply dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: I finally found what I remembered: #1146 (https://jira.duraspace.org/browse/DS-2542). That PR handles timestamps in SOLR but is not related to this one.

Comment (marsaoua) [reply dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: To ease a review: 
These changes corrects the syntaxes and requirements of Solr as we use it. If you take a look on how locations are indexed here: https://github.com/DSpace/DSpace/blob/716912f275b233522dcab982d049e0f01b9a0b60/dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java#L694 you will notice that a "m" or "l" is added to the ID of Community/Collection.
As for the lastModified field, Solr doesn't support comparison operators such as "<=, =>, ..."  for Dates. For more information, see: https://lucene.apache.org/solr/guide/6_6/working-with-dates.html

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java [hunk 2]
```diff
@@ -12,13 +12,15 @@
 import java.text.MessageFormat;
 import java.text.ParseException;
 import java.util.ArrayList;
+import java.util.Calendar;
```
Comment (Generalelektrix) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: This import is reported as never used.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java [hunk 3]
```diff
@@ -12,13 +12,15 @@
...
 import java.util.Date;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.ResourceBundle;
+import java.util.TimeZone;
```
Comment (Generalelektrix) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: This import is reported as never used.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java [hunk 4]
```diff
@@ -267,17 +269,23 @@ else if ( ! itemsChanged(context, dso, cFeed.timeStamp))
...
-
-        // convert dates to ISO 8601, stripping the time
-        String startDate = dcStartDate.toString().substring(0, 10);
-        String endDate = dcEndDate.toString().substring(0, 10);
-        
+     
         // this invocation should return a non-empty list if even 1 item has changed
         try {
-            return (Harvest.harvest(context, dso, startDate, endDate,
+            return (Harvest.harvest(context, dso, dcStartDate.toString(), dcEndDate.toString(),
```
Comment (Generalelektrix) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: I'm not sure about this. I'm going to first test a deployment without editing FeedServlet.java. The thing is, startDate and endDate are passed as strings to harvest, but then they are re-instanciated into DCDate objects. We want granularity = DateGran.DAY here, so better keep it this way.

Comment (Generalelektrix) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: Ok, after a better code review, this looks correct. I will be deploying this locally later this week and report afterwards.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java (hunk 1 above)
Comment (Generalelektrix) [reply dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java]: Code deployed in production here. It fixes the error messages we had in dspace.log and RSS feeds are correctly displaying.

**Comment by tdonohue** [comment]
FYI, @Generalelektrix noted in the ticket comments that he tested this PR (both in dev and production) and found that it fixed the errors: https://jira.duraspace.org/browse/DS-3768?focusedCommentId=59334&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-59334

**Comment by kshepherd** [comment]
noting this PR (excluding the webui specific commits) should be ported to master

**Review by kshepherd — APPROVED** [review]
+1 RSS feeds working, and sub-daily scripts don't report errors (when merged with #1891 )
as noted, this will need a partial port to master.
suggest others test this merged with #1891 as it relies on this PR

**Review by tdonohue — APPROVED** [review]
👍 Code looks reasonable to me. Seems to do what is described. Also well tested from others.

**Comment by kshepherd** [comment]
Merging along with #1891. Thanks @marsaoua !

**Pull request merged**